File: DurakApi/Models/Room.cs
namespace DurakApi.Models
{
    public class Room
    {
        public Guid Id {  get; private set; }
        public List<User> Users { get; set; } = [];
        public bool IsPlaying { get; private set; } = false;

        public Room() 
        { 
            Id = Guid.NewGuid();
        }

        public bool CanPlay => Users.Count > 1;
    }
}

File: durak_app/lib/models/api/room.dart
import 'package:durak_app/models/api/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'generated/room.g.dart';

@JsonSerializable()
class Room {
  String id;
  String name;
  bool isPlaying;
  List<User> users;

  Room({
    required this.id,
    required this.name,
    required this.users,
    this.isPlaying = false,
  });

  factory Room.fromJson(Map<String, dynamic> json) => _$RoomFromJson(json);

  // Map<String, dynamic> toJson() => _$RoomToJson(this);
  Map<String, dynamic> toJson() => { "id": "a"  };
}
